src/movies.js

```diff
@@ -1,22 +1,55 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const mappedArray = movies.map(function (singleMovie){
+    return singleMovie.director;
+   })
+   return mappedArray;
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies){
+  const movieDirector = movies.filter( (directorName) => {
+    return directorName.director === 'Steven Spielberg';
+  })
+  for (let i=0; movieDirector === 'Steven Spielberg'; i++){
+    const directorDrama = movieDirector.filter( (dramaMovie) => {
+      return dramaMovie.genre === 'Drama';
+    })
+  }
+  return directorDrama;
 
+}
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  let movieScores = movies.reduce(function (accValue, currentValue){
+    return accValue.score + currentValue.score
+  })
```

Comment:
My suggestion is to go back to lectures in student portal, 'Arrays & Objects'  specifically, you have detailed example there on how to use reduce method on array of objects. 
1.  when using reduce with objects, you always need to define initial value, add zero usually:

  let movieScores = movies.reduce(function (accValue, currentValue){
    return accValue.score + currentValue.score
  }, 0)

2. Since accValue is an initial value that you set yourself, it is an integer (not an object), and has no properties That's why you get that error, ''accValue.score' was not defined'.

Reply:
Ohh I see, I've read it again